### PR TITLE
fix view resolving for stream

### PR DESCRIPTION
**Pull request by SamuelBrucksch**
Until now views were not resolved for stream queries. This is handled now correctly. Requires a new version of @sap/cds though.

Should not be merged until that version is available.

### CODE REVIEW

File: db-service/lib/SQLService.js [hunk 1]
```diff
@@ -176,7 +176,12 @@ class SQLService extends DatabaseService {
     const cqn = this.cqn4sql(q)
 
     const cmd = cqn.cmd || Object.keys(cqn)[0]
-    if (cmd in { INSERT: 1, DELETE: 1, UPSERT: 1, UPDATE: 1 }) {
+    if (cmd in { INSERT: 1, DELETE: 1, UPSERT: 1, UPDATE: 1, STREAM: 1 }) {
+      if (cmd === 'STREAM' && cqn[cmd].from) {
+        // only resolve stream insert
+        return new this.class.CQN2SQL(this.context).render(cqn, values)
+      }
+
```
Comment (sjvans) [inline db-service/lib/SQLService.js:184]:
```suggestion
    if (cmd in { INSERT: 1, DELETE: 1, UPSERT: 1, UPDATE: 1 } || cqn.STREAM?.into) {
```
wouldn't this do the same?

**Review by sjvans — CHANGES_REQUESTED** [review]
see https://github.com/cap-js/cds-dbs/pull/96/files#r1257918662

**Review by sjvans — APPROVED** [review]
🏂

**Review by BobdenOs — APPROVED** [review]

**Comment by johannes-vogel** [comment]
we should also increase the peer dependency version as streaming does not work without the cds fix, shouldn't we?

File: db-service/package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "lint": "npx eslint . && npx prettier --check . "
   },
   "peerDependencies": {
-    "@sap/cds": ">=7"
+    "@sap/cds": ">=7.1.0"
```
Comment (sjvans) [inline db-service/package.json:35]:
```suggestion
    "@sap/cds": ">=7.1.1"
```

File: package-lock.json [hunk 3]
```diff
@@ -1248,12 +1248,12 @@
       }
     },
     "node_modules/@sap/cds": {
-      "version": "7.0.0",
-      "resolved": "https://registry.npmjs.org/@sap/cds/-/cds-7.0.0.tgz",
-      "integrity": "sha512-eQhAVVvS6ZaOdGDhdkH1YhRbNjTMF+C/xIkWxkOiklZmr77T8sa685Lxh6PRRSs5aAVlK+A/hJF0cgk9Y8NQrg==",
+      "version": "7.1.0",
```
Comment (sjvans) [inline package-lock.json:1251]: update needed

File: db-service/CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@
 - `UPDATE` is only noop if it does not include an element annotated with `@cds.on.update`.
 - `SELECT` with `'*'` that is not expanded creates now a clearer error when the column name is required.
 - `SELECT` with plain SQL statements will return correct result regardless of casing.
+- View resolving for streams
```
Comment (sjvans) [inline db-service/CHANGELOG.md:15]:
```suggestion
- View resolving for streams.
```

**Pull request merged**